src/gtk/toga_gtk/widgets/box.py

```diff
@@ -37,28 +37,29 @@ def do_get_preferred_height(self):
 
     def do_size_allocate(self, allocation):
         # print(self._impl, "Container layout", allocation.width, 'x', allocation.height, ' @ ', allocation.x, 'x', allocation.y)
-        self.set_allocation(allocation)
-        self.interface.refresh()
+        if self._impl.viewport is not None:
```

Comment:
Can you help me understand why we are using self._impl at the implementation level? I thought that was being used to call the implementation from the interface.

Reply:
In this case, we're actually working on the *native* object (`TogaBox`); `_impl` is the reference to the implementation object (which just happens to be defined in the same file)